fsharp-backend/src/ApiServer/Api/APISecrets.fs

```diff
@@ -92,3 +92,73 @@ module Delete =
                 { secret_name = s.name; secret_value = s.value })
               secrets }
     }
+
+module InsertV1 =
+
+  type Secret = { name : string; value : string }
+
+  type Params = Secret
+
+  type T = { secrets : List<Secret> }
+
+  /// API endpoint to insert a Secret within a canvas
+  let insert (ctx : HttpContext) : Task<T> =
+    task {
+      use t = startTimer "read-api" ctx
+      try
+        t.next "read-api"
+        let canvasInfo = loadCanvasInfo ctx
+        let! p = ctx.ReadVanillaJsonAsync<Params>()
+        Telemetry.addTags [ "secretName", p.name ]
+
+        t.next "insert-secret"
+        do! LibBackend.Secret.insert canvasInfo.id p.name p.value
+
+        t.next "get-secrets"
+        let! secrets = LibBackend.Secret.getCanvasSecrets canvasInfo.id
+
+        t.next "write-api"
+        return
+          { secrets =
+              List.map
+                (fun (s : PT.Secret.T) -> { name = s.name; value = s.value })
+                secrets }
+      with
+      | e ->
+        let msg = e.ToString()
+
+        if String.includes "duplicate key value violates unique constraint" msg then
+          Exception.raiseEditor
+            "The secret's name is already defined for this canvas"
+        else
+          e.Reraise()
+
+        return { secrets = [] }
+    }
+
+module DeleteV1 =
```

Comment:
considering the UI doesn't expose the functionality, I'd be tempted to remove both DeletedV0 and DeletedV0, and the unused reference in API.res

Maybe a cleanup item? again, not blocking for the PR

Reply:
It's coming next!